sklearn/decomposition/dict_learning.py

```diff
@@ -56,6 +57,7 @@ def _sparse_encode(X, dictionary, gram, cov=None, algorithm='lasso_lars',
         omp: uses orthogonal matching pursuit to estimate the sparse solution
         threshold: squashes to zero all coefficients less than regularization
         from the projection dictionary * data'
+        linear: uses a non-penalized least square fit (regularization parameter is ignored)
```

Comment:
Actually this will not yield a sparse code, so I wonder if adding this option here does not reflect a design problem.